# Challenge Week 1 Pandas
By: Alex Guerrero

Challenge Questions by @LLCdatainmotion
1. Which was the most-ordered item?
2. For the most-ordered item, how many items were ordered?
3. What was the most ordered item in the choice_description column?
4. How many items were ordered in total?
5. Turn the item price into a float
6. How much was the revenue for the period in the dataset?
7. How many orders were made in the period?
8. What is the average revenue amount per order?
9. How many different items are sold?

In [1]:
#import package
import pandas as pd

In [2]:
#Read in .tsv file as .csv:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'
chipotle = pd.read_csv(url, sep = '\t')

## Exploring Data

In [3]:
chipotle.head() #display first five rows

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [4]:
chipotle.tail() #display last five rows

order_id  quantity           item_name  \
4617      1833         1       Steak Burrito   
4618      1833         1       Steak Burrito   
4619      1834         1  Chicken Salad Bowl   
4620      1834         1  Chicken Salad Bowl   
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

In [5]:
chipotle.shape #dimension of dataframe

(4622, 5)

In [6]:
chipotle.dtypes

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

In [7]:
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


----
### 1. Which was the most-ordered item?

In [7]:
df1 = chipotle.groupby('item_name').sum('quantity') #grouping by item name and summing up quantities
df1 = df1.sort_values(by='quantity', ascending=False) #sorting values desc order
df1.head()

order_id  quantity
item_name                              
Chicken Bowl           713926       761
Chicken Burrito        497303       591
Chips and Guacamole    449959       506
Steak Burrito          328437       386
Canned Soft Drink      304753       351

A: The chichen bowl was the most-ordered item.

----
### 2. For the most-ordered item, how many items were ordered?
A: The chicken bowl was the most-ordered item with 761 orders.

----
### 3. What was the most ordered item in the choice_description column?

In [ ]:
#choice_description column
#extract each order entry
#go through entry and get all items within


In [8]:
chipotle['choice_description']

0                                                     NaN
1                                            [Clementine]
2                                                 [Apple]
3                                                     NaN
4       [Tomatillo-Red Chili Salsa (Hot), [Black Beans...
                              ...                        
4617    [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...
4618    [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...
4619    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...
4620    [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...
4621    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...
Name: choice_description, Length: 4622, dtype: object

In [15]:
choices_list = []
for choices in chipotle['choice_description']:
    # for choice in choices:
    #     choices_list.append(choice)
    print(choices)

nan
[Clementine]
[Apple]
nan
[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]
[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]
nan
[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]
[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]
[Fresh Tomato Salsa, [Rice, Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]
nan
[Roasted Chili Corn Salsa, [Fajita Vegetables, Rice, Black Beans, Cheese, Sour Cream]]
[Roasted Chili Corn Salsa, [Rice, Black Beans, Cheese, Sour Cream]]
[Fresh Tomato Salsa, [Fajita Vegetables, Rice, Cheese, Sour Cream, Guacamole]]
nan
nan
[Tomatillo-Green Chili Salsa (Medium), [Pinto Beans, Cheese, Sour Cream]]
[Fresh Tomato Salsa (Mild), [Black Beans, Rice, Cheese, Sour Cream, Lettuce]]
[Sprite]
[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Sour Cream, Cheese, Lettuce]]
nan
[[Fresh Tomato Salsa (Mild), Tomatillo-Gr

----
### 4. How many items were ordered in total?

----
### 5. Turn the item price into a float

In [9]:
chipotle['item_price'] = chipotle['item_price'].str.replace('$','') #replacing '$' with blank
chipotle['item_price'] = chipotle['item_price'].astype(float) #assigning type as float

<ipython-input-9-cc22c40b5f00>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  chipotle['item_price'] = chipotle['item_price'].str.replace('$','') #replacing '$' with blank


In [10]:
chipotle.dtypes #confirming type

order_id                int64
quantity                int64
item_name              object
choice_description     object
item_price            float64
dtype: object

In [11]:
chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98

----
### 6. How much was the revenue for the period in the dataset?

In [12]:
total_revenue = chipotle['item_price'].sum()
print(f"The total revenue for the time period of the dataset is ${total_revenue}.")

The total revenue for the time period of the dataset is $34500.16.


----
### 7. How many orders were made in the period?

In [13]:
#get count of unique order id
total_count_order = len(chipotle['order_id'].unique()) #got array of unique order_id
print(f'During this period there were {total_count_order} total orders made.')


During this period there were 1834 total orders made.


----
### 8. What is the average revenue amount per order?

In [14]:
avg_per_order = round(total_revenue/total_count_order,2) #rounding the avg into 2 decimals
print(f'The average revenue amount per order is ${avg_per_order}.')

The average revenue amount per order is $18.81.


----
### 9. How many different items are sold?